src/pip/_internal/utils/models.py

```diff
@@ -5,10 +5,20 @@
 
 import operator
 
+from pip._internal.utils.typing import MYPY_CHECK_RUNNING
 
-class KeyBasedCompareMixin(object):
+if MYPY_CHECK_RUNNING:
+    from typing import List
+
+
+class Base(object):
+    __slots__ = []  # type: List[str]
+
+
+class KeyBasedCompareMixin(Base):
     """Provides comparison capabilities that is based on a key
     """
+    __slots__ = ['_compare_key', '_defining_class']
```

Comment:
I think you missed the newline above of this.

Reply:
```suggestion

    __slots__ = ['_compare_key', '_defining_class']
```